GeoDaCenter/chicago-environment-explorer
# Feature: Index Builder

## Problem
As a User, I should be able to select from a list of predetermined indicators and choose a weight for each indicator.

This would allow me to build up a custom index to use for viewing the map for my own unique analysis

First (almost complete) draft of #119 

## Approach
* Added `scaling` and `ignore_0` to Data Dictionary and Variables
    * `scaling` tells us whether higher values are good (1) or bad (-1)
    * `ignore_0` tells us whether 0 is a valid value for this variable (false) or whether we should exclude 0 values from the calculations (true)

Current order of operations:
1. Loop over all geoJson features and for each selected indicator, determine mean and sampleStandardDeviation, use mean and standard deviation to compute zScore for each value
2. For each geoJson feature, apply weights as a percentage of zScore value, and accumulate sum in `CUSTOM_INDEX`
3. After weighted sums are accumulated, we should now have all `CUSTOM_INDEX` values - use these to determine min/max and scale from 0 to 1
4. Use scaled values to determine quantile bins - right now the code assumes there are always 3 partitions (aka 4 bins)
5. Now that we have our scaled values and the quantile bins, insert/update a pseudo-variable in variablePresets and display a map where we’ve selected this variable + Legend with a (hard-coded) colorScale

## How to Test
1. Navigate to `https://deploy-preview-120--chicago-env-explorer.netlify.app/builder`
2. Click on the lightgreen (i) symbol to see more information about a particular indicator
3. Select one or more Indicators by clicking the boxes on the right 
4. Click on Next at the bottom to see the Weights page
5. Slide to or type in the weight for each indicator
6. Click on Next at the bottom to see the Summary & Map page
    * You should see your Custom Index is visualized on the map
    * You should see that Custom Index values are scaled from 0 to 1
